### PR TITLE
DM-25156: Declare curated calibration types in instrument class, not config

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: python/lsst/obs/subaru/_instrument.py [hunk 1]
```diff
@@ -55,6 +55,8 @@ class HyperSuprimeCam(Instrument):
     policyName = "hsc"
     obsDataPackage = "obs_subaru_data"
     filterDefinitions = HSC_FILTER_DEFINITIONS
+    additionalCuratedDatasetTypes = ("bfKernel", "transmission_optics", "transmission_sensor",
+                                     "transmission_filter", "transmission_atmosphere", "yBackground")
```
Comment (kfindeisen) [inline python/lsst/obs/subaru/_instrument.py:59]: It seems a bit odd that we've all been doing HSC work without needing this info before. Doesn't `ci_hsc` use these datasets? Is there a config somewhere else that needs to be updated?

Comment (timj) [reply python/lsst/obs/subaru/_instrument.py:59]: These were all listed explicitly in the obs_base default config for conversion...

**Comment by timj** [comment]
What else did you expect to find in obs_subaru?

File: python/lsst/obs/subaru/_instrument.py (hunk 1 above)
Comment (RobertLuptonTheGood) [reply python/lsst/obs/subaru/_instrument.py:59]: HSC doesn't separate these terms (although it'll be nice to have them for LSSTCam).  One point: I'm not sure that `transmission_atmosphere` belongs here as it's per-visit.  Also, it's a little odd to have `transmission_sensor` which is probably meant to be `1 - QE`, but it's a little unclear in the IR (around 950nm and redder) as the CCD becomes transparent.
Who defined this set of parameters?

Comment (timj) [reply python/lsst/obs/subaru/_instrument.py:59]: These are the calibrations that the HyperSuprimeCam gen3 instrument writes to the calibration collection when it is asked to write curated calibrations.

Comment (RobertLuptonTheGood) [reply python/lsst/obs/subaru/_instrument.py:59]: Someone must have chosen them, though.  And they don't make a lot of sense;  I'd be sorry to see them get baked into gen3 or HSC.

Comment (TallJimbo) [reply python/lsst/obs/subaru/_instrument.py:59]: My thinking was to leave this as-is until Gen2 is gone, not use Gen3 to intentionally break it.

**Pull request merged**